_episodes/02-image-basics.md

```diff
@@ -97,50 +60,141 @@ color, but that each pixel can have a different color from its neighbors.
+
+ ~~~
+eight_image = skimage.io.imshow(eight)
+~~~
+{: .language-python}
+
+While images may also contain descriptive metadata that is essential in many instances, the bulk of a picture file is just arrays of numeric infomation that when interpreted according to the correct rule set becomes recognizable as an image to us.  Thus if we have tools that will allow us to manipulate these arrays of numbers, we can manipulate the image.  Let's try that out using numpy array slicing. Notice that the default behavior of the imageshow function appended row and an column numbers that will be helpful to us as we try to address indiviual or groups of pixels. First lets' make a copy of our eight, then make it look like a zero.
+
+To make it look like a zero, we need to change the number underlying the center most pixel to be 1.  With the help of those row and column headers, at this small scale we can determine  the center pixel is in row labled 2 and column labeled 1. Using array slicing, we can then address and assign a new value to that position.
+
+ ~~~
+zero = np.loadtxt('data/eight.csv',delimiter=',')
+zero[2,1]= 1.
+zero_image = skimage.io.imshow(zero)
+~~~
+{: .language-python}
+> ## Changing Pixel Values (5 min)
+>
+> Make another copy of eight named five, and then 
+> change the value of pixels so you have a five in drawn in black pixels.
+>
+> > ## Solution
+> >There are many possible solution, but one method would be . . .
+> {: .solution}
+{: .challenge}
 
-When we process images, we can access, examine, and / or change the color of
-any pixel we wish. To do this, we need some convention on how to access pixels
-individually; a way to give each one a name, or an address of a sort.
+## More colors
+Up to now, we only had a 2 color matrix, but we can have more if we use other numbers or fractions. One common way is to use the numbers between 0 and 255 to allow for 256 different colors or 256 different levels of grey.  Let's try that out.
+ ~~~
+#make a copy of eight
+three_colors = eight[:]
 
-The most common manner to do this, and the one we will use in our programs,
-is to assign a modified Cartesian coordinate system to the image. The
-coordinate system we usually see in mathematics has a horizontal x-axis and
-a vertical y-axis, like this:
+#multiply the whole matrix by 128
+three_colors = three_colors * 128
 
-![Cartesian coordinate system](../fig/01-cartesian.png)
+# set the middle row (index 2) to the value of 255., so you end up with the values 0.,128.,and 255
+three_colors[2,:] = 255.
 
-The modified coordinate system used for our images will have only positive
-coordinates, the origin will be in the upper left corner instead of the
-center, and y coordinate values will get larger as they go down instead of
-up, like this:
+skimage.io.imshow(three_colors)
```

Comment:
We might include the output of imshow in the lesson pages; but a few examples might help both instructors and learners while going through the steps, i.e., learners can cross-check their output with what is on the course page as a confirmation that "it works'.

Reply:
> Add images for the code output.
Just saw that this is already covered.